Projects/Scripts/Engines/Spawner/Spawner.cs

```diff
@@ -472,49 +473,59 @@ public bool Spawn(SpawnerEntry entry, out EntryFlags flags)
           else
             paramargs = entry.Parameters.Trim().Split(' ');
 
-          ConstructorInfo[] ctors = type.GetConstructors();
-
-          for (int i = 0; i < ctors.Length; ++i)
+          if (paramargs.Length == 0)
+          {
+            o = ActivatorUtil.CreateInstance(type, ci => Add.IsConstructible(ci, AccessLevel.Developer));
+          }
+          else
           {
-            ConstructorInfo ctor = ctors[i];
+            ConstructorInfo[] ctors = type.GetConstructors();
 
-            if (Add.IsConstructible(ctor, AccessLevel.Developer))
+            for (int i = 0; i < ctors.Length; ++i)
             {
-              ParameterInfo[] paramList = ctor.GetParameters();
+              ConstructorInfo ctor = ctors[i];
 
-              if (paramargs.Length == paramList.Length)
+              if (Add.IsConstructible(ctor, AccessLevel.Developer))
               {
-                object[] paramValues = Add.ParseValues(paramList, paramargs);
+                ParameterInfo[] paramList = ctor.GetParameters();
 
-                if (paramValues != null)
+                if (paramargs.Length == paramList.Length)
                 {
-                  o = ctor.Invoke(paramValues);
-                  for (int j = 0; j < realProps.Length; j++)
-                    if (realProps[j] != null)
-                    {
-                      object toSet = null;
-                      string result = Properties.ConstructFromString(realProps[j].PropertyType, o,
-                        props[j, 1], ref toSet);
-                      if (result == null)
-                      {
-                        realProps[j].SetValue(o, toSet, null);
-                      }
-                      else
-                      {
-                        flags = EntryFlags.InvalidProps;
-
-                        (o as ISpawnable)?.Delete();
-
-                        return false;
-                      }
-                    }
-
-                  break;
+                  object[] paramValues = Add.ParseValues(paramList, paramargs);
+
+                  if (paramValues != null)
+                  {
+                    o = ctor.Invoke(paramValues);
+                    break;
+                  }
                 }
               }
             }
           }
 
+          for (int j = 0; j < realProps.Length; j++)
+          {
+            if (realProps[ j ] != null)

```

Comment:
While we don't have formal linting, I try to stick to no spaces between index accessors.